File: latihan a/hardware.java
class hardware extends produk
{ 
/*kelas yang digunakan untuk mengimplementasikan hardware/
/*atribut hardware*/
		private String brand;		//brand
		private String model;		//model
	
		public hardware(){
		/*costructor*/
		}
		
		public hardware(String a, String b){
		/*costructor*/
			this.brand = a;
			this.model = b;
		}
	
	/*mengeset nilai brand*/
	public void setbrand(String a)
	{
		this.brand = a;
	}
	/*mengeset nilai model*/
	public void setmodel(String b)
	{
		this.model = b;
	}

	/*mengembalikan nilai brand*/
	String getbrand()
	{
		return this.brand;
	}
	/*mengembalikan nilai model*/
	String getmodel()
	{
		return this.model;
	}
}

File: latihan b/Latihan2 C++/Hardware.cpp
#include <iostream>
#include <string>

using namespace std;

// import file kelas parent
#include "Produk.cpp"

class Hardware : public Produk{ 

/*kelas yang digunakan untuk mengimplementasikan Hardware/
/*atribut Hardware*/
		private:
			string brand = "";		//brand
			string model = "";		//model
	
		public:
		
		Hardware(){
		/*costructor*/
		}
		
		 // constructor dengan parameter
        Hardware(string brand, string model){
            this->brand = brand;
            this->model = model;
        }
	
	/*mengeset nilai brand*/
	void setbrand(string brand)
	{
		this->brand = brand;
	}
	/*mengeset nilai model*/
	void setmodel(string model)
	{
		this->model = model;
	}

	/*mengembalikan nilai brand*/
	string getbrand()
	{
		return this->brand;
	}
	/*mengembalikan nilai model*/
	string getmodel()
	{
		return this->model;
	}
	
	~Hardware(){
		/*desstructor*/
		}
};